runtime/lua/vim/lsp.lua

```diff
@@ -565,8 +565,9 @@ local function buf_attach(bufnr)
             client.request_sync(ms.textDocument_willSaveWaitUntil, params, 1000, ctx.buf)
           if result and result.result then
             util.apply_text_edits(result.result, ctx.buf, client.offset_encoding)
-          elseif err then
-            log.error(vim.inspect(err))
+          elseif err or result and result.err then
```

Comment:
Nit but I prefer using parentheses instead of relying on the precedence of `and`/`or` here.